src/storage/operations/stat.rs

```diff
@@ -0,0 +1,51 @@
+use crate::error::Result;
+use opendal::{EntryMode, Operator};
+
+#[derive(Debug, Clone)]
+pub struct ObjectMeta {
+    pub path: String,
+    pub entry_type: String, // file | dir | other
+    pub size: u64,
+    pub last_modified: Option<String>,
+    pub etag: Option<String>,
+    pub content_type: Option<String>,
+}
+
+pub trait Stater {
+    fn new(operator: Operator) -> Self;
+    async fn stat(&self, path: &str) -> Result<ObjectMeta>;
+}
+
+#[derive(Clone)]
+pub struct OpenDalStater {
+    operator: Operator,
+}
+
+impl Stater for OpenDalStater {
+    fn new(operator: Operator) -> Self {
+        Self { operator }
+    }
+
+    async fn stat(&self, path: &str) -> Result<ObjectMeta> {
```

Comment:
Change `stat(&self, path: &str)` to `stat<P: AsRef<str>>(&self, path: P)` to improve usability.

- More flexible entry types accept `&str`, `String`, `Box<str>`, `Cow<'_, str>`, `Arc<String>`, etc. as long as `AsRef<str>` is available.
- Users don't need to manually `.as_str()` or fetch references, making calls smoother.
- Reduce unnecessary copy/allocation internally, use `path.as_ref()` to get `&str`, avoid turning `String` again or cloning.
- More efficient for temporary strings, spliced results, etc. (zero-cost abstraction).